worlds/mm-recomp/Rules.py

```diff
@@ -296,14 +330,191 @@ def get_location_rules(player):
+            lambda state: state.has("Hookshot", player),
+            
+            
+        "Zora Cape Near Great Fairy Grotto Chest":
+            lambda state: state.has("Goron Mask", player) or has_explosives(state, player),
+        "Zora Cape Underwater Like-Like HP":
+            lambda state: state.has("Zora Mask", player),
+            
+            
+        "Great Bay Temple Gekko Entrance Room Pot SF":
+            lambda state: state.has("Great Fairy Mask", player),
+        "Great Bay Temple Green Pipe Lever Room Underwater Barrel SF":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Four Torches Chest":
+            lambda state: can_use_fire_arrows(state, player),
+        "Great Bay Temple Wart Chest":
+            lambda state.has("Progressive Bow", player),
+        "Great Bay Temple Gekko Entrance Room Caged Chest":
+            lambda state: can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player),
+        "Great Bay Temple Red-Green Pipe First Room Chest":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Green-Yellow Pipe Chest":
+            lambda state: can_use_ice_arrows(player, state),
+        "Great Bay Temple Green Pipe Freezable Waterwheel Upper Chest":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Green Pipe Freezable Waterwheel Lower Chest":
+            lambda state: can_use_ice_arrows(state, player),
+        "Great Bay Temple Green Pipe Lever Room Chest":
+            lambda state: can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player),
+        "Great Bay Temple Heart Container":
+            lambda state: (can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player) and state.has("Small Key (Great Bay Temple)", player) and state.has("Boss Key (Great Bay Temple)", player)) or (state.has("Gyorg's Remains", player) and state.has("Progressive Bow", player)),
+        "Great Bay Temple Gyorg's Remains":
+            lambda state: (can_use_ice_arrows(state, player) and can_use_fire_arrows(state, player) and state.has("Small Key (Great Bay Temple)", player) and state.has("Boss Key (Great Bay Temple)", player)) or (state.has("Gyorg's Remains", player) and state.has("Progressive Bow", player)),
+        
 
         "Road to Ikana Pillar Chest":
             lambda state: state.has("Hookshot", player),
         "Road to Ikana Rock Grotto Chest":
             lambda state: state.has("Goron Mask", player),
         "Road to Ikana Stone Soldier":
             lambda state: can_play_song("Epona's Song", state, player) and has_bottle(state, player) and state.has("Progressive Magic Upgrade", player) and state.has("Lens of Truth", player),
+            
+            
+        "Graveyard Day 1 Chest":
+            lambda state: state.has("Captain's Hat", player) and can_smack_hard(state, player) and can_use_fire_arrows(state, player),
```

Comment:
Lighting the torches does not spawn the chest. However it opens the door leading to the Iron Knuckle, that needs to be defeated to obtain the Song of Storms check

Reply:
Shifted fire arrows down to "Song of Storms" rule.